exchanges/binance/binance_websocket.go

```diff
@@ -307,13 +206,85 @@ func (b *Binance) WsHandleData() {
 				}
 
 				currencyPair := currency.NewPairFromString(depth.Pair)
-				b.Websocket.DataHandler <- wshandler.WebsocketOrderbookUpdate{
+				b.Websocket.DataHandler <- monitor.WebsocketOrderbookUpdate{
 					Pair:     currencyPair,
-					Asset:    "SPOT",
+					Asset:    orderbook.Spot,
 					Exchange: b.GetName(),
 				}
 				continue
 			}
 		}
 	}
 }
+
+// SeedLocalCache seeds depth data
+func (b *Binance) SeedLocalCache(p currency.Pair) error {
+	var newOrderBook orderbook.Base
+	formattedPair := exchange.FormatExchangeCurrency(b.Name, p)
+	orderbookNew, err := b.GetOrderBook(
+		OrderBookDataRequestParams{
+			Symbol: formattedPair.String(),
+			Limit:  1000,
+		})
+	if err != nil {
+		return err
+	}
+
+	for _, bids := range orderbookNew.Bids {
```

Comment:
Much prefer using an index here as it's more efficient, some of these orderbook ranges can be quite large like in this one (max of 1000 items)